Automattic/woocommerce-subscriptions-core
# Remove the REST API and webhook files and from the autoloader

### Description

Removes the REST API feature from `subscriptions-core`.

### To test: 

1. Git clone `woocommerce-subscriptions-core` into wp-plugins and checkout this branch (this will make WC Subscriptions 4.0+ and WC Payments load `subscriptions-core` from this repo)
1. **Deactivate** the Subscriptions extension
2. [Create REST API credentials](https://docs.woocommerce.com/document/woocommerce-rest-api/#section-2) for your site.
2. Using a tool like Postman send a subscription-related request.
   - eg **` GET`** `http://example.local/wp-json/wc/v3/subscriptions/{ID}`.
3. On `trunk`, the request will go through and you should receive a valid response. 
4. On this branch you will get the following error response. 


```json
{
    "code": "rest_no_route",
    "message": "No route was found matching the URL and request method.",
    "data": {
        "status": 404
    }
}
```

**Webhooks**

1. Git clone `woocommerce-subscriptions-core` into wp-plugins and checkout this branch (this will make WC Subscriptions 4.0+ and WC Payments load `subscriptions-core` from this repo)
1. **Deactivate** the Subscriptions extension
1. Go to **WooCommerce > Settings > Advanced > Webhooks**.
1. With WC Subscriptions active or on `trunk`,  you can still create subscription webhooks. https://d.pr/i/7vJIfz
1. On this branch there will be no subscription-related webhook topics. https://d.pr/i/JjjzCR

closes #3
